doc/Design.tex

```diff
@@ -10,6 +10,25 @@ \chapter{Design}
 \section{Message Manager}
 \label{sec:design-message-manager}
 
+A CubedOS application may contian one or more domains. Each domain has a Message Manager instance and one or more modules. All modules exist inside a domain. A domain is one executable. Domains can communicate through special Transport Modules which will be explained later. This design along with the message encoding scheme enables distributed CubedOS systems.
+
+The Message Manager is instiated by supplying a static description of domain it manages. This includes the domain's assigned ID and a list of what modules are present in the domain. The manager uses this information to do sanity checks during system startup, ensuring that all modules are present and ready to handle messages before any messages are sent. On startup, each module alerts the manager of its presence by submitting its module ID. If the same ID is submitted multiple times, an error is raised immediately. This allows SPARK to check for module cardinality.
+
+To read from their mailbox, modules use a object generated by their MXDR description and validated during startup by the manager. This object contains metadata about the module including a list of every message type that the module is prepared to receive. The provides a reliable interface for the module which the manager uses to provide runtime safety. Messages sent to a module will be discarded if they're of a type not on the list. The objects are also privately held by modules, making it impossible for other modules to read from the incorrect mailbox.
+
+The manager provides checked and unchecked send operations for messages. The finite size of mailboxes demands that messages destined for a full mailbox be discarded. Checked send operations will alert the user immediately if a message is discarded for this reason, while unchecked operations will not.
+
```

Comment:
Added to manual:
- Explanation of domains
- Explanation of message safety, generated API files